common/remote/sockfile.cpp

```diff
@@ -2885,25 +2967,32 @@ class CAsyncCommandManager
 
 
 
-inline void appendErr(MemoryBuffer &reply, RemoteFileCommandType e)
+inline void appendErr(MemoryBuffer &reply, unsigned e)
 {
-    reply.append((unsigned)e).append(getRFCText(e));
+    reply.append(e).append(getRFSERRText(e));
 }
-inline void appendErr2(MemoryBuffer &reply, RemoteFileCommandType e, unsigned v)
+inline void appendErr2(MemoryBuffer &reply, unsigned e, unsigned v)
 {
     StringBuffer msg;
-    msg.append(getRFCText(e)).append(':').append(v);
-    reply.append((unsigned)e).append(msg.str());
+    msg.append(getRFSERRText(e)).append(':').append(v);
+    reply.append(e).append(msg.str());
+}
+inline void appendErr3(MemoryBuffer &reply, unsigned e, int code, const char *errMsg)
+{
+    StringBuffer msg;
+    msg.appendf("ERROR: %s(%d) '%s'", getRFSERRText(e), code, errMsg?errMsg:"");
+    reply.append(e);
+    reply.append(msg.str());
 }
-inline void appendErr3(MemoryBuffer &reply, RemoteFileCommandType e, int code, const char *errMsg)
+inline void appendCmdErr(MemoryBuffer &reply, unsigned e, int code, const char *errMsg)
 {
     StringBuffer msg;
     msg.appendf("ERROR: %s(%d) '%s'", getRFCText(e), code, errMsg?errMsg:"");
-    // some errors are RemoteFileCommandType, some are RFSERR_*
     // RFCOpenIO needs remapping to non-zero for client to know its an error
+    // perhaps we should use code here instead of e ?
     if ((RemoteFileCommandType)e == RFCopenIO)
-        e = (RemoteFileCommandType)RFSERR_OpenFailed;
-    reply.append((unsigned)e);
+        e = RFSERR_OpenFailed;
+    reply.append(e);
```

Comment:
For RFCopenIO it maps specifically to an appropriate error here.

As it says in throttleCommand, most errors should have been caught and dealt with by here.
It might be sensible to add a RFSERR_Unknown, for use here. The client will have the exception text anyway.
Passing the command back here as the error code isn't really right (not new).